src/main/java/httpApplication/Application.java

```diff
@@ -0,0 +1,8 @@
+package httpApplication;
+
+import httpserver.HttpIncomingMessage;
+import httpserver.HttpResponse;
+
+public interface Application {
+    public HttpResponse start(HttpIncomingMessage request);
```

Comment:
What is an `HttpIncomingMessage`?

Reply:
Interface for `HttpRequest` and it's mock class

Reply:
Is there a reason you didn't call it an `HttpRequest`? Further, it might be a little premature to introduce an interface if this is truly a value object/data structure object that doesn't contain behavior.

Reply:
It's legacy code from when the HttpRequest had behavior so the mock was necessary. Since it's just holding data now, I can remove it.